x-pack/solutions/security/plugins/security_solution/public/siem_migrations/common/components/migration_steps/macros/sub_steps/macros_file_upload/index.tsx

```diff
@@ -27,7 +27,7 @@ export const useMacrosFileUploadStep = ({
   missingMacros,
   onMacrosCreated,
 }: RulesFileUploadStepProps): EuiStepProps => {
-  const { upsertResources, isLoading, error } = useUpsertResources(onMacrosCreated);
+  const { upsertResources, isLoading, error } = useUpsertResources(onMacrosCreated, 'rule');
```

Comment:
This is suppose to be a **common** component, so I'm not sure why we use `'rule'` here. How that will work within the dashboards?

Also, I see that we are referencing some rules specific classes `RuleResourceIdentifier` and `RuleMigrationTaskStats`. This should be replaced with the common/abstract interfaces?